### PR TITLE
Store num draws and daily charge determination

### PR DESCRIPTION
**Pull request by iach526526**
-  daily charge:
  - More intuitive judgment of date differences
- Store :
  - Multiple draws at once 

## Summary by CodeRabbit

- **New Features**
	- Introduced a structured JSON schema for validating product data, enhancing data integrity and usability.
	- Added a new `url` field for direct links to products, improving accessibility of product information.

- **Documentation**
	- Added formal schemas for product and MediaWiki database table definitions, promoting clarity and consistency in data management.

- **Chores**
	- Enhanced the structure and readability of product data, ensuring standardized categories and logical arrangement of fields.
	- Streamlined role management functionality and improved user experience in gift handling features.
	- Optimized channel update process to reduce unnecessary operations and improve performance.

### CODE REVIEW

**Review by Edit-Mr — CHANGES_REQUESTED** [review]
照兔子說的吧。

**Comment by Edit-Mr** [comment]
晚點我 UI 再改一下

**Comment by iach526526** [comment]
資料庫欄位和表格名稱的大寫好像在這次的變更中全部更改為全小寫了，確定全部都有改到嗎？資料庫那邊我要停機做修改，然後釋出模板表格

**Comment by iach526526** [comment]
我在現在跑中電喵的伺服器上測試，大小寫是有差別的耶
MySQL-result

**Comment by winstonsung** [comment]
🤔

**Comment by winstonsung** [comment]
我試一下。

**Comment by winstonsung** [comment]
https://dev.mysql.com/doc/refman/8.4/en/identifier-case-sensitivity.html

**Comment by winstonsung** [comment]
> How table and database names are stored on disk and used in MySQL is affected by the lower_case_table_names system variable. lower_case_table_names can take the values shown in the following table. This variable does not affect case sensitivity of trigger identifiers. On Unix, the default value of lower_case_table_names is 0. On Windows, the default value is 1. On macOS, the default value is 2.

**Comment by winstonsung** [comment]
看來需要跑rename了。

**Comment by winstonsung** [comment]
看是不是剛好把資料表名稱normalize一下？

**Comment by iach526526** [comment]
是可以不用改回大寫了，資料庫可以順便整理。但其他有呼叫到的表格或欄位沒有改乾淨，我現在看就有好幾個沒有一起變小寫

**Comment by winstonsung** [comment]
那我code search一下吧。

**Comment by winstonsung** [comment]
應該取代好了（用的Visual Studio Code專案搜尋/資料夾檔案搜尋），你再檢查一下。

File: static/table_struct.sql [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  `loveuwu` tinyint(1) NOT NULL DEFAULT '0',
+  `point` int NOT NULL DEFAULT '0',
+  `ticket` int NOT NULL DEFAULT '1',
+  `charge_combo` int NOT NULL DEFAULT '0',
+  `next_lottery` int NOT NULL DEFAULT '0',
+  `last_charge` datetime NOT NULL DEFAULT '1970-01-01 00:00:00',
+  `last_comment` date NOT NULL DEFAULT '1970-01-01',
+  `today_comments` int NOT NULL DEFAULT '0',
+  PRIMARY KEY (`uid`)
+) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
```
[the remark this answers is not in the document]
Comment (iach526526) [reply static/table_struct.sql:58]: 只是先備份一下現在的表格規劃而已，兔子你別慌

File: app.py [hunk 2]
```diff
@@ -217,42 +217,51 @@ def buy_product():
...
+        user_tickets -= product["price"]
     # 更新抽獎券和電電點
+    if not discord_user:
+        return "請重新登入"
     write(discord_user["id"], "ticket", user_tickets, cursor)
     write(discord_user["id"], "point", user_points, cursor)
     end(connection, cursor)
-    return [ "抽獎成功", slot_json["get"][result], result ]
+    easter_egg=f"你是好駭客，破例讓你連抽 {num_draws} 次" if num_draws not in (1,10) else ""#給用 burp suite 偷改前端表單的人一點驚喜
+    return [ easter_egg+"抽獎成功", slot_json["get"][result], result ]
```
[the remark this answers is not in the document]
Comment (iach526526) [reply app.py]: ```cursor.execute("UPDATE game SET lastid = %s", (message.author.id,))```
這行是哪個檔案要改？

**Comment by iach526526** [comment]
@winstonsung 好，謝謝你。我明天停機維護會把資料庫那邊也一起修正

**Review by Edit-Mr — APPROVED** [review]
LGTM

File: static/switch-btn.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+	changeMachine('#1e90ff', '#00bfff', '#87cefa', '#4682b4');
+}
+
+function switchRight(){//右邊，10連
+	switchBtnRight.classList.add('active-case');
+	switchBtnLeft.classList.remove('active-case');
+	activeSwitch.style.left = '50%';
+	numDraws.value=10;
+	changeMachine('#FF0000', '#FF4500', '#B22222', '#8B0000');
+	achine('#FFD700', '#FFA500', '#FF6347', '#d76a1d');
```
[the remark this answers is not in the document]
Comment (winstonsung) [reply static/switch-btn.js:42]: CC @Edit-Mr

**Pull request merged**